### PR TITLE
extensions: invalidate progress deadline less than minreadyseconds

### PR DESCRIPTION
**Pull request by 0xmichalis**
@kubernetes/deployment ptal

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36241)

### CODE REVIEW

File: pkg/apis/extensions/validation/validation.go [hunk 1]
```diff
@@ -264,6 +264,9 @@ func ValidateDeploymentSpec(spec *extensions.DeploymentSpec, fldPath *field.Path
 	}
 	if spec.ProgressDeadlineSeconds != nil {
 		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*spec.ProgressDeadlineSeconds), fldPath.Child("progressDeadlineSeconds"))...)
+		if *spec.ProgressDeadlineSeconds <= spec.MinReadySeconds {
```
Comment (mfojtik) [inline pkg/apis/extensions/validation/validation.go]: is this breaking compatibility?

Comment (kargakis) [inline pkg/apis/extensions/validation/validation.go]: Nope, progressDeadlineSeconds for old deployments is nil so we never get to do this check.

**Comment by kargakis** [comment]
@k8s-bot test this issue: https://github.com/kubernetes/kubernetes/issues/36381

**Comment by saad-ali** [comment]
Release-czar approved post-code freeze merge--This was LGTMed and in the merge-queue at code freeze time for 1.5. Adding 1.5 milestone to let it gets merged after code freeze.

**Pull request merged**